src/main/java/tool/designpatterns/verifiers/multiclassverifiers/decorator/DecoratorVerifier.java

```diff
@@ -0,0 +1,226 @@
+        Feedback allElementsChild = patternInstance.hasAllElements(patternInstance);
+        List<Feedback> childFeedbacks = new ArrayList<>();
+        childFeedbacks.add(allElementsChild);
+        if (!allElementsChild.getIsError()) {
+            childFeedbacks.add(interfaceContainsMethod(patternInstance.interfaceComponent));
+            patternInstance.abstractDecorators.forEach(decorator -> {
+                childFeedbacks.add(hasFieldOfType(decorator, patternInstance.interfaceComponent));
+                childFeedbacks.add(
+                    fieldInitializedInConstr(decorator, patternInstance.interfaceComponent));
+            });
+            patternInstance.concreteDecorators.forEach(decorator -> {
+                childFeedbacks.add(
+                    fieldInitializedInConstr(decorator, patternInstance.interfaceComponent));
+            });
+        }
+
+        return Feedback.getPatternInstanceFeedback(childFeedbacks);
+    }
+
+    /**
+     * Checks that an interfaceComponent contains at least one public method.
+     *
+     * @param interfaceComponent The interface of the interfaceComponent
+     *
+     * @return A {@link Feedback} object containing the result
+     */
+    public Feedback interfaceContainsMethod(
+        ClassOrInterfaceDeclaration interfaceComponent) {
+        Feedback result;
+        MethodDeclaration erroringMethod = null;
+
+        AtomicBoolean resultBool = new AtomicBoolean(false);
+        for (MethodDeclaration methodDeclaration : interfaceComponent.findAll(
+            MethodDeclaration.class)) {
+            if (!methodDeclaration.isPrivate()) {
+                resultBool.set(true);
+                erroringMethod = methodDeclaration;
+            }
+        }
+
+        if (resultBool.get()) {
+            result = Feedback.getSuccessfulFeedback();
+        } else {
+            String msg = "Interface does not contain any methods!";
+            if (erroringMethod == null) {
+                result = Feedback.getPatternInstanceNoChildFeedback(msg);
+            } else {
+                result = Feedback.getNoChildFeedback(msg, new FeedbackTrace(erroringMethod));
+            }
+        }
+
+        return result;
+    }
+
+    /**
+     * Method for checking that a class contains a field of a certain type.
+     *
+     * @param toTest The class to check
+     * @param type   The class or interface that toTest should contain
+     *
+     * @return A {@link Feedback} object containing the result
+     */
+    @SuppressWarnings("PMD.LinguisticNaming")
+    private Feedback hasFieldOfType(
+        ClassOrInterfaceDeclaration toTest, ClassOrInterfaceDeclaration type) {
+        Feedback result;
+        AtomicBoolean hasAComponent = new AtomicBoolean(false);
+        toTest.findAll(VariableDeclarator.class).forEach(fieldDeclaration -> {
+            if (fieldDeclaration.getTypeAsString().contains(
+                type.getNameAsString())) { //Check that this works
```

Comment:
Why not simply use .equals()? I tested this now and .equals should do the job.
![waa](https://user-images.githubusercontent.com/31534278/77680522-5d564800-6f94-11ea-89a7-664f75494707.png)

Reply:
Equals should do the trick, good suggestion!